driver-core/src/test/java/com/datastax/driver/core/CustomTypeTest.java

```diff
@@ -22,26 +22,85 @@
 import java.util.List;
 
 import org.testng.annotations.Test;
-import static org.testng.Assert.assertEquals;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.DataType.cint;
+import static com.datastax.driver.core.DataType.custom;
+import static com.datastax.driver.core.DataType.list;
 
 /**
  * Test we "support" custom types.
  */
 public class CustomTypeTest extends CCMBridge.PerClassSingleNodeCluster {
 
+    public static final DataType CUSTOM_DYNAMIC_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.DynamicCompositeType("
+        + "s=>org.apache.cassandra.db.marshal.UTF8Type,"
+        + "i=>org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType CUSTOM_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.CompositeType("
+            + "org.apache.cassandra.db.marshal.UTF8Type,"
+            + "org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType LIST_OF_INTS = list(cint());
+
     @Override
     protected Collection<String> getTableDefinitions() {
         return Collections.singleton(
-              "CREATE TABLE test ("
-            + "    k int,"
-            + "    c 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
-            + "    v int,"
-            + "    PRIMARY KEY (k, c)"
-            + ") WITH COMPACT STORAGE"
+            "CREATE TABLE test ("
+                + "    k int,"
+                + "    c1 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
+                + "    c2 'ReversedType(CompositeType(UTF8Type, Int32Type))'," // reversed translates to CLUSTERING ORDER BY DESC
+                + "    c3 'ListType(Int32Type)'," // translates to list<int>
```

Comment:
It looks like cassandra 1.2.19 can't handle this:

```
ERROR [Native-Transport-Requests:78] 2015-11-19 07:26:14,539 ErrorMessage.java (line 210) Unexpected exception during request
java.lang.IllegalStateException: Composite column is already fully constructed
        at org.apache.cassandra.db.marshal.CompositeType$Builder.add(CompositeType.java:300)
        at org.apache.cassandra.db.marshal.CompositeType$Builder.add(CompositeType.java:332)
        at org.apache.cassandra.db.marshal.CompositeType$Builder.add(CompositeType.java:269)
        at org.apache.cassandra.cql3.Lists$Appender.doAppend(Lists.java:356)
        at org.apache.cassandra.cql3.Lists$Setter.execute(Lists.java:271)
        at org.apache.cassandra.cql3.statements.UpdateStatement.mutationForKey(UpdateStatement.java:265)
        at org.apache.cassandra.cql3.statements.UpdateStatement.getMutations(UpdateStatement.java:140)
        at org.apache.cassandra.cql3.statements.ModificationStatement.execute(ModificationStatement.java:118)
        at org.apache.cassandra.cql3.QueryProcessor.processStatement(QueryProcessor.java:103)
        at org.apache.cassandra.cql3.QueryProcessor.process(QueryProcessor.java:113)
        at org.apache.cassandra.transport.messages.QueryMessage.execute(QueryMessage.java:87)
        at org.apache.cassandra.transport.Message$Dispatcher.messageReceived(Message.java:287)
        at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
        at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
        at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
        at org.jboss.netty.handler.execution.ChannelUpstreamEventRunnable.doRun(ChannelUpstreamEventRunnable.java:43)
        at org.jboss.netty.handler.execution.ChannelEventRunnable.run(ChannelEventRunnable.java:67)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

This happens on first insert of row c3.  It looks like it *should* work.  I think it's the definition of c1 and/or c2 that is really the problem.  c1 and c2 also don't appear in system.schema_columns:

```
cqlsh> select keyspace_name, columnfamily_name, column_name, validator from system.schema_columns where keyspace_name='ks_1';

 keyspace_name | columnfamily_name | column_name | validator
---------------+-------------------+-------------+-------------------------------------------------------------------------------------
          ks_1 |              test |          c3 | org.apache.cassandra.db.marshal.ListType(org.apache.cassandra.db.marshal.Int32Type
```